.github/workflows/Build.yaml

```diff
@@ -45,13 +45,41 @@ jobs:
       - name: Check spotless
         run: ./gradlew spotlessCheck --init-script gradle/init.gradle.kts --no-configuration-cache
 
+      - name: Check Dependency Guard
+        id: dependencyguard_verify
+        continue-on-error: true
+        run: ./gradlew dependencyGuard
+
+      - name: Prevent updating Dependency Guard baselines if this is a fork
+        id: checkfork_dependencyguard
+        continue-on-error: false
+        if: steps.dependencyguard_verify.outcome == 'failure' && github.event.pull_request.head.repo.full_name != github.repository
+        run: |
+          echo "::error::Dependency Guard failed, please run the following to update baselines:\n" \
+               "    ./gradlew dependencyGuardBaseline" && exit 1
+
+        # Runs if previous job failed
+      - name: Generate new screenshots if verification failed and it's a PR
+        id: dependencyguard_baseline
+        if: steps.dependencyguard_verify.outcome == 'failure' && github.event_name == 'pull_request'
+        run: |
+          ./gradlew dependencyGuardBaseline
+
+      - name: Push new Dependency Guard baselines if available
+        uses: stefanzweifel/git-auto-commit-action@v5
+        if: steps.dependencyguard_baseline.outcome == 'success'
+        with:
+          file_pattern: '*/*.txt'
```

Comment:
Is it worth specifying the exact path for the dependency list file (i.e. `/app/dependencies/prodReleaseRuntimeClasspath.txt`)? Or are there other text files generated by dependencyGuard?

Reply:
There are multiple baseline files, one per module per configuration that we've set the plugin to track, so a single PR could update any or all of them at once. We can list them all out, we'd just have to keep the list up to date if we ever modify those configurations.

AFAICT there aren't any other `.txt` files in the repo outside of directories ignored by git, and there's a similar action to update screenshots which also uses the glob syntax, so this seemed reasonable. WDYT of merging as-is and changing to a static list if it causes an issue later?

Reply:
> There are multiple baseline files, one per module per configuration that we've set the plugin to track

This is not how the project is set up at this moment. DependencyGuard is currently configured only on the two app modules `:app` and `app-nia-catalog`

For the file pattern, a simpler glob like this could be better: `**/dependencies/*.txt`

Reply:
Done